cli/scripts/post-install.js

```diff
@@ -71,4 +71,4 @@ shell.sed('-i', 'from \'sinon\';', 'from \'../sinon\';', sinonChaiFilename)
 
 // copy experimental network stubbing type definitions
 // so users can import: `import 'cypress/types/net-stubbing'`
-shell.cp(resolvePkg('@packages/net-stubbing/lib/external-types.ts'), 'types/net-stubbing.ts')
+shell.cp(resolvePkg('@packages/net-stubbing/lib/external-types.d.ts'), 'types/net-stubbing.d.ts')
```

Comment:
Did you check if the source file actually exists in that location after the `postinstall`?
I'm not aware of any details and didn't check the tsconfig file yet, but there are many possible places where the generated file could be depending in the values in it.